tests/unit/lms/services/youtube_test.py

```diff
@@ -2,25 +2,87 @@
 
 import pytest
 
-from lms.services.youtube import YoutubeService, factory
+from lms.services.youtube import VideoNotFound, YoutubeService, factory
+from tests import factories
 
 
 class TestYoutubeService:
-    @pytest.mark.parametrize("enabled", (True, False, None))
-    def test_enabled(self, enabled):
-        svc = YoutubeService(enabled=enabled)
+    @pytest.mark.parametrize(
+        "enabled,api_key,expected_enabled",
+        (
+            (True, "api_key", True),
+            (True, "", False),
+            (False, "api_key", False),
+            (False, "", False),
+            (None, "api_key", False),
+            (None, "", False),
+        ),
+    )
+    @pytest.mark.usefixtures("http_service")
+    def test_enabled(self, enabled, api_key, expected_enabled, http_service):
+        svc = YoutubeService(enabled=enabled, api_key=api_key, http=http_service)
 
-        assert svc.enabled == enabled
+        assert svc.enabled == expected_enabled
+
+    @pytest.mark.usefixtures("http_service")
+    def test_video_info_raises_for_invalid_video(self, http_service):
+        svc = YoutubeService(enabled=True, api_key="api_key", http=http_service)
```

Comment:
We tend to create a fixture for the service that the rest of the tests use (but probably not the one for enabled above)


```
@pytest.fixture
def svc(self, http_service):
  return YoutubeService(enabled=True, api_key="api_key", http=http_service)
```

and then just add `svc` as parameter in these.